### ISSUE TITLE
Update definitions of all core types

### ISSUE THREAD

**Issue by paulhauner** [sigp/lighthouse#100]
## !! WAITING ON RELEASE CANDIDATE !!

This issue is presently on-hold until the specification is stable.

## Description

This is a "mega issue" which contains multiple discreet tasks. Mega issues exist to avoid cluttering the issues page a multitude of small tasks.

If you wish to work on one of these tasks, comment below and a maintainer will break the task out into a separate issue.

## Mega Issue Tasks

The definition for all structs can be found in the [Data Structures](https://github.com/ethereum/eth2.0-specs/blob/master/specs/core/0_beacon-chain.md#data-structures) section of the specification.

Types are to be defined in separate files in the `lighthouse/beacon_chain/types` crate [here](https://github.com/sigp/lighthouse/tree/master/beacon_chain/types/src).

All of these tasks _should_ be quite simple. 

- Update the `ShardAndCommittee` struct.
- Update the `SlashableVoteData` struct.
- Update the `AttestationData` struct.
- Add the `AttestationDataAndCustodyBit` struct.
- Update the `DepositInput` struct.
- Update the `BeaconBlock` struct.
- Update the `BeaconBlockBody` struct.
- Update the `BeaconState` struct.
- ~~Update the `ValidatorRecord` struct.~~ Included in #116.
- Update the `CrosslinkRecord` struct.
- Update the `ShardCommittee` struct.
- Add the `DepositRootVote` struct.
- Add the `ValidatorRegistryDeltaBlock` struct.

**Comment by paulhauner** [comment]
FYI, I just compared our `master` branch with the eth2.0-specs `master` branch and the issue description has been updated accordingly.

TLDR; I updated this.

**Comment by paulhauner** [comment]
Closing in favor of #182. Yay for a versioned spec.
